core/types/block.go

```diff
@@ -294,15 +296,24 @@ func NewBlock(header *Header, body *Body, receipts []*Receipt, hasher TrieHasher
 		}
 	}
 
-	if withdrawals == nil {
-		b.header.WithdrawalsHash = nil
-	} else if len(withdrawals) == 0 {
-		b.header.WithdrawalsHash = &EmptyWithdrawalsHash
-		b.withdrawals = Withdrawals{}
-	} else {
-		hash := DeriveSha(Withdrawals(withdrawals), hasher)
-		b.header.WithdrawalsHash = &hash
+	if config.IsIsthmus(header.Time) {
+		if withdrawals == nil || len(withdrawals) > 0 {
+			log.Crit("expected non-nil empty withdrawals operation list in Isthmus, but got: %v", body.Withdrawals)
+		}
+		b.header.WithdrawalsHash = header.WithdrawalsHash
 		b.withdrawals = slices.Clone(withdrawals)
+	} else {
+		// pre-Canyon
+		if withdrawals == nil {
```

Comment:
If you change this into an `} else if {` the diff will be smaller, as it avoids indenting the whole thing